pkg/e2e/spinup_test.go

```diff
@@ -119,7 +119,28 @@ func spinup(t testing.TB, ctx context.Context, cfg config) (chan error, error) {
 
 		cmd := c
 		g.Add(func() error {
-			err := cmd.Wait()
+			done := make(chan error)
+
+			// Lauching Wait in a goroutine and killing it if it times out.
+			// On windows it seems the executable starts successfully but Wait never returns.
+			go func() { done <- cmd.Wait() }()
+
+			timeout := time.After(5 * time.Second)
+
+			var er error
+			select {
+			case <-timeout:
+				// Timeout happened first, kill the process and print a message.
+				cmd.Process.Kill()
```

Comment:
do we need to kill the process here? From what I can see it is killed on line 154.

Reply:
@ivan-valkov Yes. The killing at line 154 is in the error callback. The thing is on Windows `cmd.Wait` hangs infinitely, even though the binary starts and stops successfully.

The only thing this change does it kills the binary after 5 seconds, no matter what. Hopefully it won't have any drawbacks, ideally we won't have tests running binaries for that long. Only downside I see is on windows cmd invoking tests will take longer which is okish for now.

Reply:
can we instead add an error to `err` if the timeout expires. Then the error callback will be called and we will do the killing - this way we have this functionality only in 1 place and we are consistent in how we do it.